src/vis_cpu/vis_cpu.py

```diff
@@ -20,66 +23,142 @@ def profile(fnc):
         return fnc
 
 
-def construct_pixel_beam_spline(bm_cube):
-    """Construct bivariate spline for pixelated beams for all antennas.
+logger = logging.getLogger(__name__)
 
-    Uses the ``scipy.interpolate.RectBivariateSpline`` function.
 
-    Parameters
-    ----------
-    bm_cube : array_like
-        Pixelized beam maps for each antenna (must be real-valued).
-        Shape: (NANT, BEAM_PIX, BEAM_PIX) if unpolarized, or
-        (NANT, NAXES, NFEEDS, BEAM_PIX, BEAM_PIX) if polarized.
+def _wrangle_beams(beam_idx, beam_list, polarized, nant, freq, interp=True):
+    """Perform all the operations and checks on the beams."""
```

Comment:
Would ideally document the arguments here, for clarity.

Reply:
Done.